docs/tutorial/where-to-store-app-data.md

```diff
@@ -0,0 +1,93 @@
+# Where to Store App Data
+
+This tutorial provides guidance on where to store user data properly depending
+on the app's type.
+
+## User data paths
+
+In Electron, locations of user data can be obtained and customized with
+`app.getPath` and `app.setPath` APIs, and there are 3 types of paths are used
+most commonly: `userData`, `localUserData` and `userCache`.
+
+### `userData`
+
+The `userData` path is where app should store persistent user data, such as
+config files and authentication information. Generally you should not write
+large files in this location as some environments may choose to synchronize
+files in this location to cloud, especially on Window.
+
+It by default points to:
+
+* `%APPDATA%/AppName` on Windows
+* `$XDG_CONFIG_HOME/AppName` or `~/.config/AppName` on Linux
+* `~/Library/Application Support/AppName` on macOS
+
+### `localUserData`
+
+On Linux and macOS `localUserData` is an alias to `userData`, on Windows it
+points to `%LOCALAPPDATA%/AppName` instead.
+
+On Windows, both `localUserData` and `userData` can be used to store persistent
+user data, the difference between them is, the `userData` can be synchronized to
+cloud, but `localUserData` can not.
+
+The convention is, if the user data takes lots of disk space, they should be
+written to `localUserData`, otherwise it can be written to `userData` instead.
+
+### `userCache`
+
+The `userCache` path is where cache data should be stored, files in this
+location should be safe to be removed without affecting app's state.
+
+On Linux it points to `$XDG_CACHE_HOME/AppName` or `~/.cache/AppName`.
+
+On macOS it points to `~/Library/Caches/AppName`.
+
+On Windows there is no specialized location to store cache data, and most apps
+write cache data to `%LOCALAPPDATA%`, so this path points to `localUserData` by
+default. If your app is writing files to both `localUserData` and `userCache`,
+you should consider changing the `userCache` path to a subdirectory of
+`localUserData` to avoid conflicts.
```

Comment:
```suggestion
* On Linux, it points to `$XDG_CACHE_HOME/AppName` or `~/.cache/AppName`.
* On macOS, it points to `~/Library/Caches/AppName`.
* On Windows, there is no specialized location to store cache data, and most apps
write cache data to `%LOCALAPPDATA%`, so this path points to `localUserData` by
default. If your app is writing files to both `localUserData` and `userCache`,
you should consider changing the `userCache` path to a subdirectory of
`localUserData` to avoid conflicts.
```